src/gpu/intel/gemm/host_scalars.hpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+* Copyright 2020-2025 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_INTEL_GEMM_HOST_SCALARS_HPP
+#define GPU_INTEL_GEMM_HOST_SCALARS_HPP
+
+#include "common/host_scalar_memory_storage.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace intel {
+namespace gemm {
+
+// Host side scalars support for scales implemented via Alpha argument
+
+template <typename ScalarType>
+status_t update_alpha_by_scale_value(float &alpha,
+        const host_scalar_memory_storage_t *scale_storage, bool is_dst) {
+    ScalarType value = 0;
+    status_t status = scale_storage->get_scalar_value(&value, sizeof(value));
+    assert(status == status::success);
+    if (status != status::success) return status;
+
+    if (is_dst) {
```

Comment:
`is_dst` and the related transformations of `alpha` can be safely moved up the call stack to reduce the amount of code required for the value extraction functions.